app/models/authentication.rb

```diff
@@ -25,4 +25,25 @@ def set_source
                        resource.try(:source_id)
                      end
   end
+
+  def update_status
+    super
+
+    if changed.include?("availability_status")
```

Comment:
this will never happen, because at least operation workers (and I believe neither cost/subswatch) never set this value